File: Missing Number/sol2.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        std::unordered_set<int> set;
        for(int i: nums){
            set.insert(i);
        }
        for(int i = 0; i<=nums.size();i++){
            if(set.count(i)==0){
                return i;
            }
        }
        return -1;
    }
};

File: Find Minimum in Rotated Sorted Array/sol2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        #bucket sort
        offset = -5000
        count = [0]*(5000*2+1)
        for i in nums:
            count[i-offset]+=1
        index = 0 
        for i in range(len(count)):
            for j in range(count[i]):
                nums[index] = i+offset
                index+=1
        return nums[0]